include/deal.II/base/data_out_base.h

```diff
@@ -1431,6 +1431,12 @@ namespace DataOutBase
     unsigned int
     n_cells() const;
 
+    /**
+     * Return the number of vertices per cell in this DataOutFilter.
+     */
+    unsigned int
+    n_vertices_per_cell() const;
```

Comment:
I wouldn't add this function - because it will work now but not for mixed meshes. Cannot you use the `Patch::reference_cell_type`?

https://github.com/dealii/dealii/blob/b22a991ea61976e141de37eca96f13e5a2a9f372/include/deal.II/base/data_out_base.h#L327